File: Programing_Basics_with_JS/examPreparation/programmingBasicsOnlineExam6And7April2019/03.OscarsWeekInCinema.js
function oscarsWeekInCinema(input) {
    let filmName = input[0];
    let typeOfRoom = input[1];
    let ticketsCount = Number(input[2]);

    let totalProfit = 0;
    
    switch(filmName) {
        case 'A Star Is Born':
            switch(typeOfRoom) {
                case 'normal': totalProfit = ticketsCount * 7.50; break;
                case 'luxury': totalProfit = ticketsCount * 10.50; break;
                case 'ultra luxury': totalProfit = ticketsCount * 13.50; break;
            } break;
        
        case 'Bohemian Rhapsody':
            switch(typeOfRoom) {
                case 'normal': totalProfit = ticketsCount * 7.35; break;
                case 'luxury': totalProfit = ticketsCount * 9.45; break;
                case 'ultra luxury': totalProfit = ticketsCount * 12.75; break;
            } break;

        case 'Green Book':
            switch(typeOfRoom) {
                case 'normal': totalProfit = ticketsCount * 8.15; break;
                case 'luxury': totalProfit = ticketsCount * 10.25; break;
                case 'ultra luxury': totalProfit = ticketsCount * 13.25; break;
            } break;
        
        case 'The Favourite':
            switch(typeOfRoom) {
                case 'normal': totalProfit = ticketsCount * 8.75; break;
                case 'luxury': totalProfit = ticketsCount * 11.55; break;
                case 'ultra luxury': totalProfit = ticketsCount * 13.95; break;
            } break;
    }

    console.log(`${filmName} -> ${totalProfit.toFixed(2)} lv.`);
}   

oscarsWeekInCinema(["A Star Is Born",
"luxury",
"42"])

oscarsWeekInCinema(["Green Book",
"normal",
"63"])

File: Programming_Basics_with_Python/programming_basics_online_exam_6_and_7_april_2019/03._oscars_week_in_cinema.py
def cinema_income(film_name, type_of_room, tickets_count, a_star_is_born, bohemian_rhapsody, green_book, the_favourite):
    total_income = 0
    if film_name == 'A Star Is Born':
        if type_of_room == 'normal':
            total_income = tickets_count * a_star_is_born['normal']
        elif type_of_room == 'luxury':
            total_income = tickets_count * a_star_is_born['luxury']
        elif type_of_room == 'ultra luxury':
            total_income = tickets_count * a_star_is_born['ultra luxury']
    elif film_name == 'Bohemian Rhapsody':
        if type_of_room == 'normal':
            total_income = tickets_count * bohemian_rhapsody['normal']
        elif type_of_room == 'luxury':
            total_income = tickets_count * bohemian_rhapsody['luxury']
        elif type_of_room == 'ultra luxury':
            total_income = tickets_count * bohemian_rhapsody['ultra luxury']
    elif film_name == 'Green Book':
        if type_of_room == 'normal':
            total_income = tickets_count * green_book['normal']
        elif type_of_room == 'luxury':
            total_income = tickets_count * green_book['luxury']
        elif type_of_room == 'ultra luxury':
            total_income = tickets_count * green_book['ultra luxury']
    elif film_name == 'The Favourite':
        if type_of_room == 'normal':
            total_income = tickets_count * the_favourite['normal']
        elif type_of_room == 'luxury':
            total_income = tickets_count * the_favourite['luxury']
        elif type_of_room == 'ultra luxury':
            total_income = tickets_count * the_favourite['ultra luxury']

    return total_income


film_name = input()
type_of_room = input()
tickets_count = int(input())

a_star_is_born = {
    'normal': 7.50,
    'luxury': 10.50,
    'ultra luxury': 13.50,
}

bohemian_rhapsody = {
    'normal': 7.35,
    'luxury': 9.45,
    'ultra luxury': 12.75,
}

green_book = {
    'normal': 8.15,
    'luxury': 10.25,
    'ultra luxury': 13.25,
}

the_favourite = {
    'normal': 8.75,
    'luxury': 11.55,
    'ultra luxury': 13.95,
}

print(f'{film_name} -> {cinema_income(film_name, type_of_room, tickets_count, a_star_is_born, bohemian_rhapsody, green_book, the_favourite):.2f} lv.')